crates/egui/src/widgets/plot/mod.rs

```diff
@@ -1350,12 +1364,18 @@ impl PreparedPlot {
             item.shapes(&mut plot_ui, transform, &mut shapes);
         }
 
-        let cursors = if let Some(pointer) = response.hover_pos() {
+        let (cursors, hovered_indexes) = if let Some(pointer) = response.hover_pos() {
             self.hover(ui, pointer, &mut shapes)
         } else {
-            Vec::new()
+            (Vec::new(), None)
         };
 
+        if hovered_indexes.is_some() {
+            self.hovered_indexes = hovered_indexes;
+        } else {
+            self.hovered_indexes = None;
+        }
+
```

Comment:
It seems that this is enougth
```suggestion
        self.hovered_indexes = hovered_indexes;

```

Reply:
Thank !